pkg/generators/system/sidekiq_deployment.go

```diff
@@ -54,12 +54,21 @@ func (gen *SidekiqGenerator) Deployment() basereconciler.GeneratorFunction {
 							{
 								Name:  gen.GetComponent(),
 								Image: fmt.Sprintf("%s:%s", *gen.ImageSpec.Name, *gen.ImageSpec.Tag),
-								Args: []string{
-									"rake",
-									"sidekiq:worker",
-									"RAILS_MAX_THREADS=25",
-								},
-								Env: pod.BuildEnvironment(gen.Options),
+								Command: func() (command []string) {
+									command = []string{"/usr/bin/bash", "-c", "bundle", "exec", "sidekiq"}
+									command = append(command, *gen.Spec.Config.QueuesArg)
+									return
+								}(),
+								Env: func() []corev1.EnvVar {
+									envVars := pod.BuildEnvironment(gen.Options)
+									envVars = append(envVars,
+										corev1.EnvVar{
+											Name:  "RAILS_MAX_THREADS",
+											Value: fmt.Sprintf("%d", *gen.Spec.Config.MaxThreads),
+										},
+									)
```

Comment:
Can't we add this to `pkg/generators/system/config/options.go` and just let this envvar be generated like all the others with `pod.BuildEnvironment()`?

Reply:
We can't, because this envvar, although is not being used right now on system-app, it can be used there and would need a different value.

Al the things we have in the `pkg/generators/system/config/options.go` are things that are common to all deployments, defined on the CR at main `.config`. This one is the first envvar that can not be shared, only beloging to one of the 3 type of deployments (app, sphinx, sidekiq x 3).

We had this issues with backend and the 3 deployment, where most of the envvars are shared from the main `.config`, but then each one has also its own `.config`, and we needed to divide the options among the 3 deployments, were almost all the vars were shared, doing a huge copy paste of most of the envvars

However on the system case, from more than 50 shared envvars, there is a single one which only applies to sidekiq, so instead of copying all of them on app + sidekiq + sphinx different option files, I preferred to stay with a single file, and add this var on the deployment directly. 

If in the future we see there are more envvars that only applies to one deployment on its own `.config`, then we will need to maybe split the options into different files, and refactor it to not having to duplicate so much code like the backend case.

Reply:
Well, there are ways to use the common options struct and add custom fields without having to copy&paste everything, but it's fine leaving it as it is now. Not worth doing that for just one single envvar.